src/structures/ButtonComponent.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+
+const BaseComponent = require('./BaseComponent');
+const { ButtonStyles, ComponentTypes } = require('../util/Constants');
+const Util = require('../util/Util');
+
+/**
+ * Represents a button component.
+ * @extends {BaseComponent}
+ */
+class ButtonComponent extends BaseComponent {
+  /**
+   * @typedef {BaseComponentOptions} ButtonComponentOptions
+   * @property {string} [label] The text to be displayed on this button
+   * @property {string} [customId] A unique string to be sent in the interaction when clicked
+   * @property {ButtonComponentStyleResolvable} [style] The style of this button
+   * @property {EmojiIdentifierResolvable} [emoji] The emoji to be displayed to the left of the text
+   * @property {string} [url] Optional URL for link-style buttons
+   * @property {boolean} [disabled=false] Disables the button to prevent interactions
+   */
+
+  /**
+   * @param {ButtonComponent|ButtonComponentOptions} [data={}] ButtonComponent to clone or raw data
+   */
+  constructor(data = {}) {
+    super({ type: 'BUTTON' });
+
+    /**
+     * The text to be displayed on this button
+     * @type {?string}
+     */
+    this.label = data.label ?? null;
+
+    /**
+     * A unique string to be sent in the interaction when clicked
+     * @type {?string}
+     */
+    this.customId = data.custom_id ?? data.customId ?? null;
+
+    /**
+     * The style of this button
+     * @type {?ButtonStyle}
+     */
+    this.style = data.style ? ButtonComponent.resolveStyle(data.style) : null;
```

Comment:
How can this be missing?

Reply:
And everything else too? `new ButtonComponent().setStyle()` thats how.

Reply:
Yeah forgot about the constructor, I thought this was the only property that would be always set